src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/CheckinDetailViewModel.swift

```diff
@@ -2,19 +2,75 @@
 // 🦠 Corona-Warn-App
 //
 
-import Foundation
+import UIKit
+import OpenCombine
 
 final class CheckinDetailViewModel {
 
 	// MARK: - Init
+	
 	init(
-		_ checkin: Checkin
+		_ traceLocation: TraceLocation
 	) {
-		self.checkin = checkin
+		self.traceLocation = traceLocation
+		
+		if let defaultDuration = traceLocation.defaultCheckInLengthInMinutes {
+			selectedDurationInMinutes = defaultDuration
+		} else {
+			let eventDuration = Calendar.current.dateComponents(
+				[.minute],
+				from: traceLocation.startDate ?? Date(),
+				to: traceLocation.endDate ?? Date()
+			)
+			// the 0 should not be possible since we expect either the defaultCheckInLengthInMinutes or the start and end dates to be available always
+			selectedDurationInMinutes = eventDuration.minute ?? 0
+		}
```

Comment:
why it it with traceLocation? Checkin should have all values require.

Reply:
This is what i understood from our call, parsing a QR code produces a TraceLocation Object which we yse for the checking details, then we use this + the user choices for checkout time and the switch value to create the check-in model which we save to the store

Reply:
What is the screen for? Editing a CheckIn or a TranceLocation? If first the model, should be a CheckIn if second, the ViewController name is confusing. You are right the QR scan will resolve to a TraceLocation.
If the screen shows an trace location you should rename it if it shows the checkin, think about creating the checkin before this screen is created and pass it in.

Reply:
I renamed the screen to TraceLocationDetailsViewController